### DISCUSSION TITLE
[Reverse Proxy] How to configure for Nginx and SN on different hosts

### DISCUSSION THREAD

**Discussion by IMOKURI** [HewlettPackard/swarm-learning#150] [Q&A] [answered]
Hello,
Thank you for adding the reverse proxy example.

In [the reverse proxy example](https://github.com/HewlettPackard/swarm-learning/tree/d0839df927c22bc399d76ec32d5c4f7e4b7bce8d/examples/reverse-proxy), the nginx and SN nodes are on the same host.
I think that nginx and SN nodes are often on different hosts.

In that case, can the `--sn-p2p-service` option of the `run-sn` command specify the host where nginx is running?
Or do I need to specify the host where the SN node is running?

The reason I ask this is because in this example, when starting SN, the P2P service name (which is usually expected to translate to the nginx IP address) is listed, but nginx does not listen on port 30303.

https://github.com/HewlettPackard/swarm-learning/blob/7772029a8ae1fa7ffc983c217dcfc9357de8dafd/examples/reverse-proxy/mnist/run-all#L178-L192

https://github.com/HewlettPackard/swarm-learning/blob/7772029a8ae1fa7ffc983c217dcfc9357de8dafd/examples/reverse-proxy/mnist/nginx.conf#L50-L55

https://github.com/HewlettPackard/swarm-learning/pull/148#issuecomment-1354258434

**Comment by girisairam450** [comment]
This example is more like a quick start for understanding swarm learning with reverse proxy. As of now sn-p2p-service still uses the 30303 port for communication, having it on the Nginx is more like a placeholder within this example. Please refer to the note highlighted in read me files of these examples. 

Yes, when you have Nginx and SN running on two different machines then you have to change the configurations. The tables mentioned in the readme in such a scenario will be different. Please try with the Bind 9 configuration where only sn-p2p-service resolves to the host where SN is running.

**Comment by IMOKURI** [comment, marked as answer]
Thank you.
I have confirmed that it works by setting the following.
(It does not seem to work when the port number is not specified.)

`--sn-p2p-service=<SN FQDN>:30303`

**Reply by girisairam450** [reply]
Yes, the 30303 Ethereum port needs to be opened explicitly. Also, We have added a multi-host example for reverse proxy. Request you to try and share your feedback. [https://github.com/HewlettPackard/swarm-learning/tree/master/examples/reverse-proxy/cifar10]
